common/eth2/src/lib.rs

```diff
@@ -686,7 +694,7 @@ impl BeaconNodeHttpClient {
     ///
     /// Returns `Ok(None)` on a 404 error.
     pub async fn post_beacon_blocks<T: EthSpec, Payload: AbstractExecPayload<T>>(
-        &self,
+        self,
```

Comment:
```suggestion
        &self,
```

This will cause a compile error elsewhere, but we can fix that by adding `let beacon_node = beacon_node;` under [this line](https://github.com/macladson/lighthouse/blob/d6184fb3c3b3990e798e9110a4a284da5ee9c50d/validator_client/src/block_service.rs#L523).

What you were achieving changing `&self` to `self` was telling the compiler to move `beacon_node` into the closure. Usually we'd do that by adding `move` to the closure, but that moves *all* variables (which doesn't compile either). Adding `let beacon_node = beacon_node;` tells the compiler to move that specific variable. I'm actually not sure if that's the idiomatic way to do it or not.. I had a look around the webz but didn't find much.